### PR TITLE
add : retryWrapper for all necessary methods in EthereumChainReader

### LINKED ISSUE #465: [contracts] Retry all calls in chainReader
**Describe the bug**
Dealing with eth nodes is often unreliable. The `chainReader` should retry these calls, especially since no gas is wasted. Ideally we'd sort by error messages, but we don't even need the implementation to be that complex.

### PR DESCRIPTION
**Pull request by jakekidd**
## The Problem
Fixes #465 
>Dealing with eth nodes is often unreliable. The chainReader should retry these calls, especially since no gas is wasted. Ideally we'd sort by error messages, but we don't even need the implementation to be that complex.

## The Solution
Add `retryWrapper` method and implement across the board. Have a constant defined amount # of retries.

### CODE REVIEW

File: modules/contracts/src.ts/services/ethReader.ts [hunk 1]
```diff
@@ -574,40 +558,61 @@ export class EthereumChainReader implements IVectorChainReader {
...
+        return Result.ok(cleaned);
+      }
+    );
+  }
+
+  private async retryWrapper<T>(
+    chainId: number,
+    targetMethod: (provider: JsonRpcProvider) => Promise<Result<T, ChainError>>
+  ): Promise<Result<T, ChainError>> {
+    // TODO : Should we retry this attempt to get provider as well?
```
Comment (LayneHaber) [inline modules/contracts/src.ts/services/ethReader.ts:602]: No we can leave this outside of the retries

File: modules/contracts/src.ts/services/ethReader.ts [hunk 2]
```diff
@@ -574,40 +558,61 @@ export class EthereumChainReader implements IVectorChainReader {
...
-      }
-    }
-    if (!registered) {
-      try {
-        registered = await registry.getTransferDefinitions();
-      } catch (e) {
-        return Result.fail(new ChainError(e.message, { chainId, transferRegistry }));
-      }
+    // TODO: Move this constant where it belongs.
+    const MAX_RETRIES = 5;
```
Comment (LayneHaber) [inline modules/contracts/src.ts/services/ethReader.ts:608]: Let's just move this to `modules/types/constants.ts` and name it something more distinct

File: modules/contracts/src.ts/services/ethReader.ts [hunk 3]
```diff
@@ -574,40 +558,61 @@ export class EthereumChainReader implements IVectorChainReader {
...
+    let res = await targetMethod(provider);
+    let retries = 0;
+
+    while (res.getError() && retries < MAX_RETRIES) {
```
Comment (LayneHaber) [inline modules/contracts/src.ts/services/ethReader.ts:612]: can use the `isError` flag here

**Review by LayneHaber — COMMENTED** [review]
Looks good, I would take care of the `TODO` for the constant and if possible add at least one test where the provider throws on the first call and resolves on the second.

File: modules/contracts/src.ts/services/ethReader.ts (hunk 2 above)
Comment (jakekidd) [reply modules/contracts/src.ts/services/ethReader.ts]: done, named `ETH_READER_MAX_RETRIES`. If that's good, pls resolve comment, thx
\

File: modules/contracts/src.ts/services/ethReader.ts (hunk 3 above)
Comment (jakekidd) [reply modules/contracts/src.ts/services/ethReader.ts]: Oooh good to know

File: modules/contracts/src.ts/services/ethReader.ts (hunk 1 above)
Comment (jakekidd) [reply modules/contracts/src.ts/services/ethReader.ts:606]: Copy

**Review by LayneHaber — APPROVED** [review]

**Pull request merged**